**In this task, I'm working with the Food Balance Sheet data from the Food and Agriculture Organization (FAO)**

In [24]:
#I would be using just Pandas for this task, hence I'd import only Pandas
import pandas as pd

In [2]:
data_df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv', sep=',', encoding='latin-1')
data_df

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

------

QUESTION 13  
WHAT IS THE TOTAL NUMBER AND PERCENTAGE OF MISSING DATA IN 2016 TO 2 DECIMAL PLACES?

In [3]:
#WHAT IS THE TOTAL NUMBER OF MISSING DATA IN 2016?
data_df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

Based on the above code, the total number of missing data in 2016 is 1535

In [4]:
#WHAT IS THE PERCENTAGE OF MISSING DATA IN 2016?
percentage_missing_data = data_df.isnull().sum() * 100 / len(data_df)
percentage_missing_data

Area Code       0.000000
Area            0.000000
Item Code       0.000000
Item            0.000000
Element Code    0.000000
Element         0.000000
Unit            0.000000
Y2014           2.607354
Y2015           2.540078
Y2016           2.518747
Y2017           2.471162
Y2018           2.356300
dtype: float64

The percentage of missing data in 2016 to 2 decimal places is 2.52

--------

QUESTION 12  
WHAT IS THE MEAN AND STANDARD DEVIATION ACROSS THE WHOLE DATASET FOR THE YEAR 2015 TO 3 DECIMAL PLACES?

In [5]:
#Drop missing values in the data
data_fao = data_df.dropna()
data_fao

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[57784 rows x 12 columns]

In [6]:
#WHAT IS THE MEAN AND STANDARD DEVIATION ACROSS THE WHOLE DATASET FOR THE YEAR 2015?
data_fao.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  57784.000000  57784.000000  57784.000000   57784.000000   57784.000000   
mean     134.499342   2690.495656   3745.422868     137.831770     138.990488   
std       72.651212    146.625825   2217.914615    1588.660769    1625.441821   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2563.000000    674.000000       0.000000       0.000000   
50%      136.000000   2633.000000   5142.000000       0.140000       0.130000   
75%      196.000000   2781.000000   5301.000000       9.000000       9.000000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   57784.000000   57784.000000   57784.000000  
mean      140.376380     144.927790     148.027664  
std      1662.732983    1695.436448    1735.920219  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.130000       0.160000       0.130000  
75%         9.000000      10.000000      10.000000  
max    185960.000000  190873.000000  195875.000000

The mean and standard deviation for the year 2015 (to 3 decimal places) are 138.990 and 1625.442 respectively

------

QUESTION 11  
WHAT IS THE TOTAL SUM OF ANIMAL FAT PRODUCED IN 2014 AND 2017 RESPECTIVELY?

In [7]:
#WHAT IS THE TOTAL SUM OF ANIMAL FAT PRODUCED IN 2014 AND 2017 RESPECTIVELY?
data_fao.groupby(['Item']).sum()

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                            
Alcohol, Non-Food        35230     693999       1389590    2329.00    2119.00   
Alcoholic Beverages      73136    1616972       2110145  102391.11   98774.72   
Animal Products          18060     397035         90990   11935.65   11811.73   
Animal fats              83742    1847142       2497151  209460.54  200675.72   
Apples and products      67924    1321585       1849486    9499.23   10559.15   
...                        ...        ...           ...        ...        ...   
Vegetables, Other        75817    1474430       2179992  155038.96  158104.08   
Vegetal Products         18060     391905         90990  107145.19  107064.17   
Wheat and products       85608    1607040       2583023  232670.13  234710.51   
Wine                     63806    1269090       1707019    4494.36    4250.81   
Yams                     39196     752895       1089374  200396.96  203150.78   

                         Y2016      Y2017      Y2018  
Item                                                  
Alcohol, Non-Food      2164.00    2299.00    2285.00  
Alcoholic Beverages   96939.75   95572.06   97842.27  
Animal Products       11661.69   11547.65   11578.61  
Animal fats          183314.22  269617.53  269648.27  
Apples and products    9850.26   10197.90    9640.51  
...                        ...        ...        ...  
Vegetables, Other    156209.90  157749.59  163978.21  
Vegetal Products     106792.72  107655.20  107775.39  
Wheat and products   239377.07  240047.62  242645.19  
Wine                   3870.09    4172.02    4035.32  
Yams                 220624.20  229172.59  221270.09  

[119 rows x 8 columns]

The total sum of animal fat produced in 2014 was 209460.54, and that of 2017 was 269617.53

----

QUESTION 14  
WHICH YEAR HAD THE HIGHEST CORRELATION WITH "ELEMENT CODE"?

In [16]:
corr_2014 = data_fao["Element Code"].corr(data_fao["Y2014"])
corr_2014

0.026257087010024112

In [17]:
corr_2015 = data_fao["Element Code"].corr(data_fao["Y2015"])
corr_2015

0.025631860410360176

In [18]:
corr_2016 = data_fao["Element Code"].corr(data_fao["Y2016"])
corr_2016

0.025080372338308714

In [19]:
corr_2017 = data_fao["Element Code"].corr(data_fao["Y2017"])
corr_2017

0.025932344579844665

In [20]:
corr_2018 = data_fao["Element Code"].corr(data_fao["Y2018"])
corr_2018

0.025988578712157227

In [21]:
#creating a list
correlation = [corr_2014, corr_2015, corr_2016, corr_2017, corr_2018]
correlation

[0.026257087010024112,
 0.025631860410360176,
 0.025080372338308714,
 0.025932344579844665,
 0.025988578712157227]

In [23]:
#WHICH YEAR HAS THE HIGHEST CORRELATION WITH ELEMENT CODE?
max(correlation)

0.026257087010024112

Year 2014 has the highest correlation (0.026257087010024112) with element code 

------

QUESTION 15  
WHAT YEAR HAD THE HIGHEST SUM OF IMPORT QUANTITY?

In [10]:
#WHAT YEAR HAD THE HIGHEST SUM OF IMPORT QUANTITY?
data_fao.groupby(["Element"]).sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   698435   13986139      27634113   
Export Quantity                            415968    7894703      17224654   
Fat supply quantity (g/capita/day)         662095   13289323       3370752   
Feed                                       144633    2933176       6012369   
Food                                       652391   13091136      25020972   
Food supply (kcal/capita/day)              659187   13227714       3256920   
Food supply quantity (kg/capita/yr)        647542   12991502       3115350   
Import Quantity                            621285   12533807      26102372   
Losses                                     274321    5422290      10286984   
Other uses (non-food)                      208464    4179786       7859850   
Processing                                 268111    5266227      10149118   
Production                                 515820   10251046      20969355   
Protein supply quantity (g/capita/day)     661878   13281782       3319450   
Residuals                                  612713   12209783      23642410   
Seed                                       103175    2025415       4189466   
Stock Variation                            569564   11284699      21378480   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996714.35  2021472.55  2044822.70   
Export Quantity                          149683.64   157195.47   151372.46   
Fat supply quantity (g/capita/day)        10225.52    10234.82    10101.07   
Feed                                     216906.89   224998.22   228949.65   
Food                                    1212296.49  1232283.10  1246953.17   
Food supply (kcal/capita/day)            454248.00   453365.00   451790.00   
Food supply quantity (kg/capita/yr)       49648.98    49341.68    48982.15   
Import Quantity                          274037.48   266841.46   286393.78   
Losses                                   153222.00   155438.00   157787.00   
Other uses (non-food)                     78656.13    66151.41    69519.68   
Processing                               282905.00   287917.00   280618.00   
Production                              1931264.75  1946992.39  1943520.15   
Protein supply quantity (g/capita/day)    11836.02    11833.44    11779.62   
Residuals                                 30149.00    30044.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58752.83    34910.99    33110.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088178.10  2161167.10  
Export Quantity                          181691.80   181067.80  
Fat supply quantity (g/capita/day)        10252.19    10257.21  
Feed                                     223685.68   233441.68  
Food                                    1258820.28  1303806.28  
Food supply (kcal/capita/day)            454658.00   455241.00  
Food supply quantity (kg/capita/yr)       48687.16    49055.02  
Import Quantity                          294212.09   287736.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91566.97    91273.97  
Processing                               292826.00   308421.00  
Production                              2030035.89  2075025.89  
Protein 

Year 2017 had the highest sum of import quantity (294212.09)

------

QUESTION 16  
WHAT IS THE TOTAL NUMBER OF SUM OF PRODUCTION IN 2014?

In [11]:
#WHAT IS THE TOTAL NUMBER OF SUM OF PRODUCTION IN 2014?
data_fao.groupby(["Element"]).sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   698435   13986139      27634113   
Export Quantity                            415968    7894703      17224654   
Fat supply quantity (g/capita/day)         662095   13289323       3370752   
Feed                                       144633    2933176       6012369   
Food                                       652391   13091136      25020972   
Food supply (kcal/capita/day)              659187   13227714       3256920   
Food supply quantity (kg/capita/yr)        647542   12991502       3115350   
Import Quantity                            621285   12533807      26102372   
Losses                                     274321    5422290      10286984   
Other uses (non-food)                      208464    4179786       7859850   
Processing                                 268111    5266227      10149118   
Production                                 515820   10251046      20969355   
Protein supply quantity (g/capita/day)     661878   13281782       3319450   
Residuals                                  612713   12209783      23642410   
Seed                                       103175    2025415       4189466   
Stock Variation                            569564   11284699      21378480   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996714.35  2021472.55  2044822.70   
Export Quantity                          149683.64   157195.47   151372.46   
Fat supply quantity (g/capita/day)        10225.52    10234.82    10101.07   
Feed                                     216906.89   224998.22   228949.65   
Food                                    1212296.49  1232283.10  1246953.17   
Food supply (kcal/capita/day)            454248.00   453365.00   451790.00   
Food supply quantity (kg/capita/yr)       49648.98    49341.68    48982.15   
Import Quantity                          274037.48   266841.46   286393.78   
Losses                                   153222.00   155438.00   157787.00   
Other uses (non-food)                     78656.13    66151.41    69519.68   
Processing                               282905.00   287917.00   280618.00   
Production                              1931264.75  1946992.39  1943520.15   
Protein supply quantity (g/capita/day)    11836.02    11833.44    11779.62   
Residuals                                 30149.00    30044.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58752.83    34910.99    33110.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088178.10  2161167.10  
Export Quantity                          181691.80   181067.80  
Fat supply quantity (g/capita/day)        10252.19    10257.21  
Feed                                     223685.68   233441.68  
Food                                    1258820.28  1303806.28  
Food supply (kcal/capita/day)            454658.00   455241.00  
Food supply quantity (kg/capita/yr)       48687.16    49055.02  
Import Quantity                          294212.09   287736.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91566.97    91273.97  
Processing                               292826.00   308421.00  
Production                              2030035.89  2075025.89  
Protein 

The total sum of production in 2014 was 1931264.75

------

QUESTION 17  
WHICH OF THESE ELEMENTS HAD THE HIGHEST SUM IN 2018?

In [12]:
#WHICH OF THESE ELEMENTS HAD THE HIGHEST SUM IN 2018?
data_fao.groupby(["Element"]).sum().sort_values(by="Y2018", axis=0, ascending=False)

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   698435   13986139      27634113   
Production                                 515820   10251046      20969355   
Food                                       652391   13091136      25020972   
Total Population - Both sexes                6020     112545         22995   
Food supply (kcal/capita/day)              659187   13227714       3256920   
Processing                                 268111    5266227      10149118   
Import Quantity                            621285   12533807      26102372   
Feed                                       144633    2933176       6012369   
Export Quantity                            415968    7894703      17224654   
Losses                                     274321    5422290      10286984   
Other uses (non-food)                      208464    4179786       7859850   
Food supply quantity (kg/capita/yr)        647542   12991502       3115350   
Residuals                                  612713   12209783      23642410   
Seed                                       103175    2025415       4189466   
Stock Variation                            569564   11284699      21378480   
Protein supply quantity (g/capita/day)     661878   13281782       3319450   
Fat supply quantity (g/capita/day)         662095   13289323       3370752   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996714.35  2021472.55  2044822.70   
Production                              1931264.75  1946992.39  1943520.15   
Food                                    1212296.49  1232283.10  1246953.17   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Food supply (kcal/capita/day)            454248.00   453365.00   451790.00   
Processing                               282905.00   287917.00   280618.00   
Import Quantity                          274037.48   266841.46   286393.78   
Feed                                     216906.89   224998.22   228949.65   
Export Quantity                          149683.64   157195.47   151372.46   
Losses                                   153222.00   155438.00   157787.00   
Other uses (non-food)                     78656.13    66151.41    69519.68   
Food supply quantity (kg/capita/yr)       49648.98    49341.68    48982.15   
Residuals                                 30149.00    30044.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58752.83    34910.99    33110.12   
Protein supply quantity (g/capita/day)    11836.02    11833.44    11779.62   
Fat supply quantity (g/capita/day)        10225.52    10234.82    10101.07   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088178.10  2161167.10  
Production                              2030035.89  2075025.89  
Food                                    1258820.28  1303806.28  
Total Population - Both sexes           1112641.00  1140605.00  
Food supply (kcal/capita/day)            454658.00   455241.00  
Processing                               292826.00   308421.00  
Import Quantity                          294212.09   287736.09  
Feed                                     223685.68   233441.68  
Export Quantity                          181691.80   181067.80  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91566.97    91273.97  
Food supply quantity (kg/capita/yr)       48687.16    49055.02  
Residual

The element with the highest sum in 2018 was Domestic supply quantity

-----

QUESTION 18  
WHICH OF THE ELEMENTS HAD THE 3RD LOWEST SUM IN 2018?

In [13]:
data_fao.groupby(["Element"]).sum().sort_values(by="Y2018", axis=0, ascending=True)

Area Code  Item Code  Element Code  \
Element                                                                      
Tourist consumption                         50308    1486528       2869905   
Fat supply quantity (g/capita/day)         662095   13289323       3370752   
Protein supply quantity (g/capita/day)     661878   13281782       3319450   
Stock Variation                            569564   11284699      21378480   
Seed                                       103175    2025415       4189466   
Residuals                                  612713   12209783      23642410   
Food supply quantity (kg/capita/yr)        647542   12991502       3115350   
Other uses (non-food)                      208464    4179786       7859850   
Losses                                     274321    5422290      10286984   
Export Quantity                            415968    7894703      17224654   
Feed                                       144633    2933176       6012369   
Import Quantity                            621285   12533807      26102372   
Processing                                 268111    5266227      10149118   
Food supply (kcal/capita/day)              659187   13227714       3256920   
Total Population - Both sexes                6020     112545         22995   
Food                                       652391   13091136      25020972   
Production                                 515820   10251046      20969355   
Domestic supply quantity                   698435   13986139      27634113   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Tourist consumption                         416.00      349.00       89.00   
Fat supply quantity (g/capita/day)        10225.52    10234.82    10101.07   
Protein supply quantity (g/capita/day)    11836.02    11833.44    11779.62   
Stock Variation                           58752.83    34910.99    33110.12   
Seed                                      21922.92    23976.82    23389.20   
Residuals                                 30149.00    30044.00    37224.00   
Food supply quantity (kg/capita/yr)       49648.98    49341.68    48982.15   
Other uses (non-food)                     78656.13    66151.41    69519.68   
Losses                                   153222.00   155438.00   157787.00   
Export Quantity                          149683.64   157195.47   151372.46   
Feed                                     216906.89   224998.22   228949.65   
Import Quantity                          274037.48   266841.46   286393.78   
Processing                               282905.00   287917.00   280618.00   
Food supply (kcal/capita/day)            454248.00   453365.00   451790.00   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Food                                    1212296.49  1232283.10  1246953.17   
Production                              1931264.75  1946992.39  1943520.15   
Domestic supply quantity                1996714.35  2021472.55  2044822.70   

                                             Y2017       Y2018  
Element                                                         
Tourist consumption                          91.00       90.00  
Fat supply quantity (g/capita/day)        10252.19    10257.21  
Protein supply quantity (g/capita/day)    11842.18    11833.43  
Stock Variation                           54334.91    20579.91  
Seed                                      24870.14    25263.14  
Residuals                                 35500.00    34864.00  
Food supply quantity (kg/capita/yr)       48687.16    49055.02  
Other uses (non-food)                     91566.97    91273.97  
Losses                                   160614.00   163902.00  
Export Quantity                          181691.80   181067.80  
Feed                                     223685.68   233441.68  
Import Quantity                          294212.09   287736.09  
Processi

The element with the 3rd lowest sum in 2018 is Protein supply quantity

-----

QUESTION 19  
WHAT IS THE TOTAL IMPORT QUANTITY IN ALGERIA IN 2018?

In [14]:
#WHAT IS THE TOTAL IMPORT QUANTITY IN ALGERIA IN 2018?
data_fao.groupby(["Element", "Area"]).sum().sort_values(by="Area", axis=0, ascending=True)

Area Code  Item Code  \
Element                             Area                             
Domestic supply quantity            Algeria         452     303390   
Losses                              Algeria         208     139857   
Other uses (non-food)               Algeria         140      95593   
Import Quantity                     Algeria         408     274912   
Processing                          Algeria         140      93537   
...                                                 ...        ...   
Production                          Zimbabwe      17557     259245   
Residuals                           Zimbabwe      18643     274860   
Food supply quantity (kg/capita/yr) Zimbabwe      19729     293228   
Import Quantity                     Zimbabwe      18462     274549   
Stock Variation                     Zimbabwe      17195     254157   

                                              Element Code     Y2014  \
Element                             Area                               
Domestic supply quantity            Algeria         599013  83755.94   
Losses                              Algeria         266396   7231.00   
Other uses (non-food)               Algeria         180390   3315.22   
Import Quantity                     Algeria         572322  34931.12   
Processing                          Algeria         179585   1776.00   
...                                                    ...       ...   
Production                          Zimbabwe        534567  16130.50   
Residuals                           Zimbabwe        532510   -252.00   
Food supply quantity (kg/capita/yr) Zimbabwe         70305    662.42   
Import Quantity                     Zimbabwe        572322   2485.58   
Stock Variation                     Zimbabwe        481840   -644.00   

                                                 Y2015     Y2016     Y2017  \
Element                             Area                                     
Domestic supply quantity            Algeria   85972.86  87157.61  86810.42   
Losses                              Algeria    7471.00   7405.00   7428.00   
Other uses (non-food)               Algeria    3483.37   3400.17   3534.08   
Import Quantity                     Algeria   37834.91  36720.08  36051.29   
Processing                          Algeria    1865.00   1920.00   2045.00   
...                                                ...       ...       ...   
Production                          Zimbabwe  14032.00  14267.60  15913.60   
Residuals                           Zimbabwe   -780.00   -882.00  -1083.00   
Food supply quantity (kg/capita/yr) Zimbabwe    627.92    620.30    615.88   
Import Quantity                     Zimbabwe   3244.03   3652.98   2083.84   
Stock Variation                     Zimbabwe   -343.00    221.00    458.00   

                                                 Y2018  
Element                             Area                
Domestic supply quantity            Algeria   90023.42  
Losses                              Algeria    7708.00  
Other uses (non-food)               Algeria    3970.08  
Import Quantity                     Algeria   36238.29  
Processing                          Algeria    2479.00  
...                                                ...  
Production                          Zimbabwe  16161.60  
Residuals                           Zimbabwe   -546.00  
Food supply quantity (kg/capita/yr) Zimbabwe    611.22  
Import Quantity                     Zimbabwe   1673.84  
Stock Variation                     Zimbabwe   -447.00  

[803 rows x 8 columns]

The total import quantity in Algeria in 2018 is 36238.29

---

QUESTION 20  
WHAT IS THE TOTAL NUMBER OF UNIQUE COUNTRIES IN THE DATASET?

In [15]:
#WHAT IS THE TOTAL NUMBER OF UNIQUE COUNTRIES IN THE DATASET?
data_fao['Area'].value_counts().count()

49

The total number of unique countries is 49